nixos/atticd.nix

```diff
@@ -146,77 +162,107 @@ in
       };
     };
   };
-  config = lib.mkIf (cfg.enable) (lib.mkMerge [
-    {
-      assertions = [
-        {
-          assertion = cfg.credentialsFile != null;
-          message = ''
-            <option>services.atticd.credentialsFile</option> is not set.
-
-            Run `openssl genrsa -traditional -out private_key.pem 4096 | base64 -w0` and create a file with the following contents:
-
-            ATTIC_SERVER_TOKEN_RS256_SECRET="output from command"
-
-            Then, set `services.atticd.credentialsFile` to the quoted absolute path of the file.
-          '';
-        }
-        {
-          assertion = !lib.isStorePath cfg.credentialsFile;
-          message = ''
-            <option>services.atticd.credentialsFile</option> points to a path in the Nix store. The Nix store is globally readable.
-
-            You should use a quoted absolute path to prevent this.
-          '';
-        }
-      ];
 
-      services.atticd.settings = {
-        database.url = lib.mkDefault "sqlite:///var/lib/atticd/server.db?mode=rwc";
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.environmentFile != null;
+        message = ''
+          <option>services.atticd.environmentFile</option> is not set.
+
+          Run `openssl genrsa -traditional -out private_key.pem 4096 | base64 -w0` and create a file with the following contents:
+
+          ATTIC_SERVER_TOKEN_RS256_SECRET="output from command"
+
+          Then, set `services.atticd.environmentFile` to the quoted absolute path of the file.
+        '';
+      }
+      {
+        assertion = !lib.isStorePath cfg.environmentFile;
+        message = ''
+          <option>services.atticd.environmentFile</option> points to a path in the Nix store. The Nix store is globally readable.
+
+          You should use a quoted absolute path to prevent leaking secrets in the Nix store.
+        '';
+      }
+    ];
 
-        # "storage" is internally tagged
-        # if the user sets something the whole thing must be replaced
-        storage = lib.mkDefault {
-          type = "local";
-          path = "/var/lib/atticd/storage";
-        };
+    services.atticd.settings = {
+      database.url = lib.mkDefault "sqlite:///var/lib/atticd/server.db?mode=rwc";
+
+      # "storage" is internally tagged
+      # if the user sets something the whole thing must be replaced
+      storage = lib.mkDefault {
+        type = "local";
+        path = "/var/lib/atticd/storage";
       };
+    };
+
+    systemd.services.atticd = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ] ++ lib.optionals hasLocalPostgresDB [ "postgresql.service" ];
+      requires = lib.optionals hasLocalPostgresDB [ "postgresql.service" ];
```

Comment:
It was pointed out to me that the after generates an eval warning. You'll want this too https://github.com/NixOS/nixpkgs/pull/349083